test/unit/lib/aws/awssigv4signer.test.js

```diff
@@ -11,6 +11,7 @@ const { test } = require('tap');
 const { URL } = require('url');
 const { v4: uuidv4 } = require('uuid');
 const AwsSigv4Signer = require('../../../../lib/aws/AwsSigv4Signer');
+const AwsSigv4SignerV3 = require('../../../../lib/aws/AwsSigv4Signer-sdk-v3');
```

Comment:
Would it be easier and less confusing if we test this in a separate file/env?

Reply:
Probably 4 different envs:
- `AwsSigv4Signer` with aws-v2
- `AwsSigv4Signer` with aws-v3
- `AwsSigv4SignerV3` with aws-v3
- `AwsSigv4SignerV3` without aws-v3

Reply:
I've moved the tests to a different file